crates/wasm-compose/tests/compositions/complex/a.wat

```diff
@@ -0,0 +1,151 @@
+    (export "o" (func $o))
+    (export "p" (func $p))
+    (export "q" (func $q))
+    (export "r" (func $r))
+    (export "s" (func $s))
+    (export "t" (func $t))
+    (export "u" (func $u))
+    (export "canonical_abi_realloc" (func $canonical_abi_realloc))
+  )
+  (core instance (instantiate 0))
+  (core alias export 0 "memory" (memory))
+  (core alias export 0 "canonical_abi_realloc" (func))
+  (core alias export 0 "a" (func))
+  (core alias export 0 "b" (func))
+  (core alias export 0 "c" (func))
+  (core alias export 0 "d" (func))
+  (core alias export 0 "e" (func))
+  (core alias export 0 "f" (func))
+  (core alias export 0 "g" (func))
+  (core alias export 0 "h" (func))
+  (core alias export 0 "i" (func))
+  (core alias export 0 "j" (func))
+  (core alias export 0 "k" (func))
+  (core alias export 0 "l" (func))
+  (core alias export 0 "m" (func))
+  (core alias export 0 "n" (func))
+  (core alias export 0 "o" (func))
+  (core alias export 0 "p" (func))
+  (core alias export 0 "q" (func))
+  (core alias export 0 "r" (func))
+  (core alias export 0 "s" (func))
+  (core alias export 0 "t" (func))
+  (core alias export 0 "u" (func))
+  (func (type 0) (canon lift (core func 1)))
+  (func (type 1) (canon lift (core func 2)))
+  (func (type 2) (canon lift (core func 3)))
+  (func (type 3) (canon lift (core func 4)))
+  (func (type 4) (canon lift (core func 5)))
+  (func (type 5) (canon lift (core func 6)))
+  (func (type 6) (canon lift (core func 7)))
+  (func (type 7) (canon lift (core func 8)))
+  (func (type 8) (canon lift (core func 9)))
+  (func (type 9) (canon lift (core func 10)))
+  (func (type 10) (canon lift (core func 11)))
+  (func (type 11) (canon lift (core func 12)))
+  (func (type 12) (canon lift (core func 13) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 14) (canon lift (core func 14) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 16) (canon lift (core func 15) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 18) (canon lift (core func 16) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 20) (canon lift (core func 17)))
+  (func (type 22) (canon lift (core func 18)))
+  (func (type 24) (canon lift (core func 19) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 27) (canon lift (core func 20) (memory 0) (realloc 0) string-encoding=utf8))
+  (func (type 29) (canon lift (core func 21) (memory 0) (realloc 0) string-encoding=utf8))
+  (export "a" (func 0))
+  (export "b" (func 1))
+  (export "c" (func 2))
+  (export "d" (func 3))
+  (export "e" (func 4))
+  (export "f" (func 5))
+  (export "g" (func 6))
+  (export "h" (func 7))
+  (export "i" (func 8))
+  (export "j" (func 9))
+  (export "k" (func 10))
+  (export "l" (func 11))
+  (export "m" (func 12))
+  (export "n" (func 13))
+  (export "o" (func 14))
+  (export "p" (func 15))
+  (export "q" (func 16))
+  (export "r" (func 17))
+  (export "s" (func 18))
+  (export "t" (func 19))
+  (export "u" (func 20))
+)
```

Comment:
Was this generated by the `wit2wasm` tool? If so it might be interesting to consider including that here perhaps

Reply:
It was generated from `cargo component` as `wit2wasm` only creates types-only components.

I simply replaced the core module with a minimal stubbed-out version by hand so we didn't have huge wat files containing rust toolchain output.

I can't think of a easier way to generate this file without dedicating tooling to it and I don't think we need it as the only maintenance needed on this file is to update it for component model / canonical ABI changes and the wat is small enough to keep that an easy task.